### PR TITLE
[LF] Change Prim to Buitlin in archive

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This is a change we wanted to do for a long time.
The idea is that we keep primitive for the surface language and builtin for LF

### CODE REVIEW

File: daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto [hunk 1]
```diff
@@ -302,23 +269,14 @@ message Type {
...
-  reserved 8; // This was contract_id. Removed in favour of PrimType.CONTRACT_ID
-  reserved 9; // This was update. Removed in favour of PrimType.UPDATE
-  reserved 10; // This was scenario. Removed in favor of PrimType.SCENARIO
-
 }
 
-// Primitive constructors
-enum PrimCon {
-
+// Builtinitive constructors
```
Comment (akrmn) [inline daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto:274]: search-and-replace leads to particitrousers :) there's handful more of these

```suggestion
// Builtin constructors
```

Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto:274]: :bow:

File: daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/iterable/ExprIterable.scala [hunk 2]
```diff
@@ -12,7 +12,7 @@ private[validation] object ExprIterable {
 
   private[iterable] def iterator(x: Expr): Iterator[Expr] = {
     x match {
-      case EVar(_) | EBuiltin(_) | EPrimCon(_) | EPrimLit(_) | EVal(_) | EEnumCon(_, _) |
+      case EVar(_) | EBuiltin(_) | EBuiltinCon(_) | EBuiltinLit(_) | EVal(_) | EEnumCon(_, _) |
```
Comment (akrmn) [inline daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/iterable/ExprIterable.scala:15]: I think we should also rename `EBuiltin`=>`EBuiltinFun` or `EBuiltinFunction`

Comment (remyhaemmerle-da) [reply daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/iterable/ExprIterable.scala:15]: make sens.

Comment (remyhaemmerle-da) [reply daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/iterable/ExprIterable.scala:15]: would you mind I make that in a follow up PR ?

File: daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto [hunk 3]
```diff
@@ -145,86 +145,53 @@ message Kind {
...
   // Builtin type 'Bool'
   BOOL = 1;
-
   // Builtin type 'Int64'
   INT64 = 2;
-
-  // Builtin type for legacy 'Decimal'
-  // Alias for (Numeric 10)
-  // *available in version < 1.7*
-  DECIMAL = 3;
```
Comment (paulbrauner-da) [inline daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto:162]: Should we also remove the ADD_DECIMAL, SUB_DECIMAL, etc. builtins in the same PR? Or maybe not remove DECIMAL in this PR.

File: daml-lf/spec/daml-lf-1.rst [hunk 4]
```diff
@@ -5123,15 +5123,15 @@ On the one hand, in case of Daml-LF 1.6 archive:
   + ``DECIMAL_TO_INT64`` message is translated to ``(NUMERIC_TO_INT64 @10)``
 
 - Numeric types, literals and builtins cannot be referred directly.
-  In other words ``numeric`` fields in ``PrimLit`` and ``PrimType``
+  In other words ``numeric`` fields in ``BuiltinLit`` and ``PrimType``
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-1.rst:5126]:
```suggestion
  In other words ``numeric`` fields in ``BuiltinLit`` and ``BuiltinType``
```

File: daml-lf/spec/daml-lf-1.rst [hunk 5]
```diff
@@ -5143,7 +5143,7 @@ On the other hand, starting from Daml-LF 1.7:
   deserialization process will reject any non-compliant program.
 
 - Decimal types, literals and builtins cannot be referred directly.
-  In other words ``decimal`` fields in ``PrimLit`` and ``PrimType``
+  In other words ``decimal`` fields in ``BuiltinLit`` and ``PrimType``
```
Comment (paulbrauner-da) [inline daml-lf/spec/daml-lf-1.rst:5146]:
```suggestion
  In other words ``decimal`` fields in ``BuiltinLit`` and ``BuiltinType``
```

File: daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto:162]: Ideally I should not have remove DECIMAL here. 
I think it is not worthwhile to revert that now.

**Review by paulbrauner-da — APPROVED** [review]

**Pull request merged**